### PR TITLE
Update signaling protocol to include new optional metric field `qpSum` and new join flags

### PR DESCRIPTION
**Pull request by XHatan**
**Issue #:**

**Description of changes:**

Update signaling protocol to include new optional metric field `qpSum`

**Testing**

1. Have you successfully run `npm run build:release` locally? yes
2. How did you test these changes? demo
3. Can these changes be tested using the demo application? If yes, which demo application can be used to test it? yes, only metrics change
4. Do you add, modify, or delete public API definitions? If yes, has that been reviewed and approved? no
5. Do you change the wire protocol, e.g. the request method? If yes, has that been reviewed and approved? it's backward compatible optional metric field change.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: protocol/SignalingProtocol.proto [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// Copyright 2019-2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (hensmi-amazon) [inline protocol/SignalingProtocol.proto:1]: Is this the new way of doing it :P ?

File: CHANGELOG.md [hunk 2]
```diff
@@ -13,6 +13,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Changed
 
+- Update `SignalingProtocol` with optional video metric fields and optional join flags
```
Comment (devalevenkatesh) [inline CHANGELOG.md:16]: Nit: Could you please add a "." at the end of this change log?

**Review by hensmi-amazon — APPROVED** [review]

**Comment by devalevenkatesh** [comment]
Could you please elaborate more on the PR question: How did you test these changes?

WIll help us while preparing for the release :)

File: protocol/SignalingProtocol.proto (hunk 1 above)
Comment (XHatan) [reply protocol/SignalingProtocol.proto:1]: good catch. I will change this.  script needs to be updated.

File: protocol/SignalingProtocol.proto [hunk 3]
```diff
@@ -246,6 +248,8 @@ message SdkMetric {
         VIDEO_DISCARDED_PPS = 47;
         VIDEO_PLIS_SENT = 48;
         VIDEO_RECEIVED_JITTER_MS = 49;
+        VIDEO_SENT_QP_SUM = 66;
```
[the remark this answers is not in the document]
Comment (XHatan) [reply protocol/SignalingProtocol.proto:251]: they should follow service protocol design.

**Comment by XHatan** [comment]
> Could you please elaborate more on the PR question: How did you test these changes?
> 
> WIll help us while preparing for the release :)

The PR is to sync up the protocol at client side with server and parse more stats at client to send to server. The change is backward compatible.

**Review by devalevenkatesh — APPROVED** [review]

**Pull request merged**